### PR TITLE
Refactor Order Details screen to be a modal

### PR DESCRIPTION
**Pull request by wachunei**

### CODE REVIEW

File: app/components/Nav/Main/MainNavigator.js [hunk 1]
```diff
@@ -131,8 +131,12 @@ const BrowserFlow = () => (
 );
 
 const TransactionsHome = () => (
-  <Stack.Navigator>
+  <Stack.Navigator mode="modal">
```
Comment (alaahd) [inline app/components/Nav/Main/MainNavigator.js:134]: This is the trick ;) where I couldn't find how to move the route outside of the feature stack.

Comment (alaahd) [reply app/components/Nav/Main/MainNavigator.js:134]: It seems even if we remove the prop `mode="modal"` we still manage to invoke the route. I can't remember why this route was failing when we took out of the original stack. It was not working from some reason.

File: app/components/Nav/Main/MainNavigator.js [hunk 2]
```diff
@@ -131,8 +131,12 @@ const BrowserFlow = () => (
...
     <Stack.Screen name="TransactionsView" component={ActivityView} />
```
Comment (alaahd) [inline app/components/Nav/Main/MainNavigator.js:135]: Do we need to set `initialRouteName` here?

File: app/components/Views/FiatOrdersView/index.js [hunk 3]
```diff
@@ -34,11 +35,8 @@ function FiatOrdersView({ orders, ...props }) {
 
   const handleNavigateToTxDetails = useCallback(
```
Comment (alaahd) [inline app/components/Views/FiatOrdersView/index.js:36]: suggest renaming this to `handleNavigateToOrderDetails`

File: app/components/UI/FiatOnRampAggregator/Views/OrderDetails.tsx [hunk 4]
```diff
@@ -20,17 +19,17 @@ const styles = StyleSheet.create({
...
   const frequentRpcList = useSelector(
     (state: any) =>
       state.engine.backgroundState.PreferencesController.frequentRpcList,
   );
-
-  const order = useSelector(makeOrderIdSelector(route.params.orderId));
-
+  const routes = useRoute();
+  // @ts-expect-error expect params error
+  const order = useSelector(makeOrderIdSelector(routes?.params?.orderId));
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/Views/OrderDetails.tsx:32]: What do you think about `getOrderIdSelector`?

File: app/components/UI/FiatOnRampAggregator/Views/OrderDetails.tsx [hunk 5]
```diff
@@ -48,10 +47,14 @@ const TransactionDetails = ({ route }: { route: any }) => {
   }, [colors, navigation]);
 
   const handleMakeAnotherPurchase = useCallback(() => {
-    // @ts-expect-error navigation prop mismatch
-    navigation.replace(Routes.FIAT_ON_RAMP_AGGREGATOR.PAYMENT_METHOD);
+    navigation.goBack();
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/Views/OrderDetails.tsx:50]: There's a type mismatch when using `pop`, we can address this later.

**Pull request merged**